src/structs.py

```diff
@@ -1001,3 +1002,9 @@ def add(self, member: Tuple[Segment, ObjToVarSub]) -> None:
 ObjToVarSub = Dict[Object, Variable]
 VarToObjSub = Dict[Variable, Object]
 Metrics = DefaultDict[str, float]
+LiftedOrGroundAtom = TypeVar(
```

Comment:
Note: it's important to do it this way instead of e.g. TypeVar("LiftedOrGroundAtom", bound=Union[LiftedAtom, GroundAtom]). Don't ask me why...

Reply:
i was curious and found this: https://stackoverflow.com/questions/59933946/difference-between-typevart-a-b-and-typevart-bound-uniona-b